src/site/content/en/blog/optimize-ttfb/index.md

```diff
@@ -0,0 +1,229 @@
+
+CDNs solve the problem of user proximity from your origin server by using a distributed network of servers that cache resources on servers that are physically closer to your users. These servers are called _edge servers_.
+
+CDN providers may also offer benefits beyond edge servers:
+
+- CDN providers usually offer extremely fast DNS resolution times.
+- A CDN will likely serve your content from edge servers using modern protocols such as HTTP/2 or HTTP/3.
+- HTTP/3 in particular solves the head-of-line blocking problem present in TCP (which HTTP/2 relies on) by using the [UDP protocol](https://en.wikipedia.org/wiki/User_Datagram_Protocol).
+- A CDN will likely also provide modern versions of TLS, which lowers the latency involved in TLS negotiation time. [TLS 1.3](https://en.wikipedia.org/wiki/Transport_Layer_Security#TLS_1.3_handshake) in particular is designed to keep TLS negotiation as short as possible.
+- Some CDN providers provide a feature often called an "edge worker", which uses an API similar to that of [the Service Worker API](https://developer.mozilla.org/docs/Web/API/Service_Worker_API) to intercept requests, programmatically manage responses in edge caches, or rewrite responses altogether.
+- CDN providers are very good at optimizing for compression. Compression is tricky to get right on your own, and may lead to slower response times in certain cases with dynamically generated markup, which must be compressed on the fly.
+- CDN providers will also automatically cache compressed responses for static resources, leading to the best mix of compression ratio and response time.
+
+While adopting a CDN involves a varying amount of effort from trivial to significant, it should be a high priority to pursue in optimizing your TTFB if your website is not already using one.
+
+### Avoid multiple page redirects
+
+One potential contributor to a high TTFB is [redirects](https://developer.mozilla.org/docs/Web/HTTP/Redirections). Redirects occur when a navigation request for a document receives a response that informs the browser that the resource exists at another location. One redirect can certainly add unwanted latency to a navigation request, but it can certainly get worse if that redirect points to another resource that results in _another_ redirect—and so on.
+
+There are two types of redirects:
+
+- **Same-origin redirects**, where the redirect occurs entirely on your website.
+- **Cross-origin redirects**, where the redirect occurs initially on another origin—such as from a social media URL shortening service, for example—before arriving at your website.
+
+You want to focus on eliminating same-origin redirects, as this is something you will have direct control over. This would involve checking links on your website to see if any of them result in a 302 or 301 response code. When it comes to cross-origin redirects, there's nothing you can really do to fix that source of added latency.
```

Comment:
```suggestion
You want to focus on eliminating same-origin redirects, as this is something you will have direct control over. This would involve checking links on your website to see if any of them result in a 302 or 301 response code.

Cross-origin redirects are trickier as these are often outside of your control, but try to avoid multiple redirects where possible - for example, by using multiple link shorteners when sharing links.
```